nh_food_and_fluid/models/food_fluid_review.py

```diff
@@ -176,6 +187,151 @@ def schedule_review(self, spell_activity):
             }
         )
 
+    @api.model
+    def add_reviews_to_periods(self, patient_id, periods):
+        """
+        Add review task data to the periods dictionaries passed from
+        get_periods_dictionaries on the F&F model
+        :param patient_id: ID for the patient we're looking up reviews for
+        :type patient_id: int
+        :param periods: list of periods
+        :type periods: list of dicts
+        :return: list of periods with review data added
+        """
+        for period in periods:
+            period['reviews'] = {}
+            for review_time in self.trigger_times:
+                period['reviews'][review_time] = self.get_review_data(
+                    patient_id, period.get('period_start_datetime'),
+                    review_time)
+        return periods
+
+    @api.model
+    def get_review_data(self, patient_id, period_start, trigger_time):
+        """
+        Get data for the review in a given period for a given patient
+        :param patient_id: Id of the patient we're looking up review for
+        :type patient_id: int
+        :param period_start: start of the period
+        :type period_start: Odoo datetime string
+        :param trigger_time: time the review was triggered (hour of day)
+        :type trigger_time: int
+        :return: dictionary of review score, state and user/reason
+        """
+        food_fluid_model = \
+            self.env['nh.clinical.patient.observation.food_fluid']
+        review_activity, obs = self.get_data_for_trigger_time(
+            patient_id, period_start, trigger_time)
+        if not review_activity:
+            return {}
+        fluid_intake = food_fluid_model.\
+            _calculate_total_fluid_intake_from_obs_activities(obs)
+        score = food_fluid_model.calculate_period_score(fluid_intake)
+        user_or_reason = self.get_user_or_reason(review_activity)
+        state = review_activity.state
+        if state == 'cancelled':
+            state = 'Not Completed'
+        return {
+            'score': score,
+            'state': state.title(),
+            'user': user_or_reason
+        }
+
+    def get_review_end(self, period_start, trigger_time):
+        """
+        Get the datetime 1 minute before the next review so we can use this
+        to find observations and reviews that fall within this timeframe
+        :param period_start: Start of the current F&F period
+        :param trigger_time: Time of the review we want the end datetime for
+        :return: Odoo datetime string representation
+        """
+        # find the next trigger time
+        trigger_time_index = self.trigger_times.index(trigger_time)
+        sorted_trigger_times = self.trigger_times[trigger_time_index:] + \
+            self.trigger_times[:trigger_time_index]
+        next_trigger_time = sorted_trigger_times[1]
```

Comment:
I'm not sure modulo would work. Say I've got 8 different times and I need to get the forth one. 

My current approach would shift the array to start at the 3rd one (the 4th one then being `sorted_trigger_times[1]`). However if we did the modulo approach you suggested we'd have `3 % 8` which in my Python console is returning `3` . 

I think the correct way to deal with the need to rotate with modulo would be `((trigger_time_index + 1) % len(self.trigger_times))-1` as if we're looking at the last item in the `trigger_times` list we'll get be doing `(9 % 8) = 1` then taking away `1` to get the correct index BUT this does not work with indexes within the length of the `trigger_times` list (i.e. `((4 + 1) % 8) -1) = 4`) where we'd want the item at index 5.

Hope that makes sense